src/Ledger/Ratify.lagda

```diff
@@ -108,25 +110,34 @@ restrictedDists coins rank dists = dists -- record dists { drepStakeDistr = rest
         restrict : Credential ↛ Coin → Credential ↛ Coin
         restrict dist = topNDRepDist rank dist ∪ᵐˡ mostStakeDRepDist dist coins
 
-votedHashes : Vote → ((GovRole × Credential) ↛ Vote) → GovRole → ℙ Credential
-votedHashes v votes r = (votes ⦅ r ,-⦆)⁻¹ v
+votedHashes : Vote → (Voter ↛ Vote) → GovRole → ℙ Voter
+votedHashes v votes r = votes ⁻¹ v
 
-votedYesHashes : ((GovRole × Credential) ↛ Vote) → GovRole → ℙ Credential
+votedYesHashes : (Voter ↛ Vote) → GovRole → ℙ Voter
 votedYesHashes = votedHashes Vote.yes
 
-votedAbstainHashes : ((GovRole × Credential) ↛ Vote) → GovRole → ℙ Credential
+votedAbstainHashes : (Voter ↛ Vote) → GovRole → ℙ Voter
 votedAbstainHashes = votedHashes Vote.abstain
 
-getStakeDist : GovRole → StakeDistrs → Credential ↛ Coin
+autoVoterStakeDist : StakeDistrs → Voter ↛ Coin
+autoVoterStakeDist record
+  { noConfidenceStake = ncs
+  ; abstainerStake = as
+  } = fromListᵐ $ (noConfidence , ncs) ∷ (abstainer , as) ∷ []
+
+getStakeDist : GovRole → StakeDistrs → Voter ↛ Coin
```

Comment:
I think this should return something of type `VDeleg ↛ Coin`. It's strange that there's a `GovRole` in there.